enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/schedule/Delay.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.core.consensus.schedule;
+
+import java.util.concurrent.TimeUnit;
+
+public class Delay
+{
+    private final long amount;
+    private final TimeUnit unit;
+
+    Delay( long amount, TimeUnit unit )
```

Comment:
Does this have advantages over using a `java.time.Duration`?

Reply:
I had some impedance mismatch trying to use Duration, because it was too flexible compared to what I would be able to support. Perhaps it was just my lack of skill dealing properly with time though.